src/main/java/com/avispl/symphony/dal/avdevices/wirelesspresentation/mersive/solsticepodgen3/SolsticePodGen3Communicator.java

```diff
@@ -0,0 +1,1256 @@
+				}
+			} else {
+				if (!property.isControl()) {
+					stats.put(propertyName, SolsticeConstant.NONE);
+				}
+			}
+		}
+	}
+
+	/**
+	 * Sends a POST request to the Solstice Pod with the specified command, property, and value.
+	 *
+	 * @param command the command to be sent in the POST request
+	 * @param property the Solstice property associated with the value
+	 * @param value the value to be sent in the request body
+	 * @return the response received from the Solstice Pod as a JSON node
+	 * @throws Exception if an error occurs during the POST request
+	 */
+	private void sendPostRequest(String command, SolsticePropertiesList property, Object value) {
+		try {
+			JsonNode response = this.doPost(command, createBodyRequest(property.getApiGroupName(), property.getApiPropertyName(), value), JsonNode.class);
+			if (response.has(SolsticeConstant.ERROR)) {
+				throw new IllegalArgumentException(
+						String.format("Can't control property %s with value %s. The device has responded with an error: %s", property.name(), value, response.get(SolsticeConstant.ERROR).asText()));
+			}
+		} catch (Exception e) {
+			throw new IllegalArgumentException(String.format("Can't control property %s with value %s. The device has responded with an error.", property.name(), value), e);
+		}
+	}
+
+	/**
+	 * Sends a command to reset the key to the device.
+	 * If an admin password is provided, it includes the password in the request.
+	 *
+	 * @throws IllegalArgumentException if the device responds with an error or if there is an issue with the request.
+	 */
+	private void sendCommandResetKey() {
+		try {
+			String request = SolsticeCommand.RESET_KEY + (this.getPassword() != null ? "?password=" + this.getPassword() : SolsticeConstant.EMPTY);
+			String response = this.doGet(request);
+			if (!SolsticeConstant.COMMAND_SUCCESSFUL.equals(response)) {
+				throw new IllegalArgumentException("The device has responded with an error.");
+			}
+		} catch (Exception e) {
+			throw new IllegalArgumentException("Can't control property ResetKey %s" + e.getMessage(), e);
+		}
+	}
+
+	/**
+	 * Sends a command to set the default background to the device.
+	 * If an admin password is provided, it includes the password in the request.
+	 *
+	 * @throws IllegalArgumentException if the device responds with an error or if there is an issue with the request.
+	 */
+	private void sendCommandSetDefaultBackground() {
+		try {
+			JsonNode response;
+			if (StringUtils.isNullOrEmpty(this.getPassword())) {
+				Map<String, String> params = new HashMap<>();
+				params.put(SolsticeConstant.FILE, SolsticeConstant.RESET_DEFAULT);
+				response = this.doPost(SolsticeCommand.SET_DEFAULT_BACKGROUND, params, JsonNode.class);
+			} else {
+				String jsonString = "{\"file\":\"reset default\",\"password\":\"" + this.getPassword() + "\"}";
+				JsonNode jsonNode = objectMapper.readTree(jsonString);
+				response = this.doPost(SolsticeCommand.SET_DEFAULT_BACKGROUND, jsonNode, JsonNode.class);
```

Comment:
I think for a structure like this, you might've used Map<String, String> for a request body